pkg/ipam/crd.go

```diff
@@ -466,6 +476,19 @@ func deriveGatewayIP(eni eniTypes.ENI) string {
 	return net.IPv4(addr[0], addr[1], addr[2], addr[3]+1).String()
 }
 
+func deriveAlibabaCloudGatewayIP(cidr string) string {
+	subnetIP, ipNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		log.WithError(err).Warningf("Unable to parse AlibabaCloud subnet CIDR %s", cidr)
+		return ""
+	}
+	addr := subnetIP.To4()
+	ones, bits := ipNet.Mask.Size()
+	// get the second last ip
+	// Ref: https://www.alibabacloud.com/help/doc-detail/65398.html
+	return net.IPv4(addr[0], addr[1], addr[2], byte(int(math.Min(math.Pow(2, float64(bits-ones)), 256))-3)).String()
+}
+
```

Comment:
There's no need to have almost the same function re-implemented. I think it's best if we try to expand the existing `deriveGatewayIP()`. It should probably accept a CIDR string and which IP to pluck out, i.e. AWS ENI is x.x.x.1 & Alibaba is x.x.x.253 (if I understood correctly). Please then create a unit test for that function